### PR TITLE
Pin to hdf5 1.8.17*

### PR DESCRIPTION
**Pull request by ocefpaf**
@kmuehlbauer I am pinning against the latest hdf5 available in conda-forge. This will be used in all packages soon. Please take a look at it and merge if you think it is OK.

Needs: https://github.com/conda-forge/netcdf-cxx4-feedstock/pull/7 and https://github.com/conda-forge/libnetcdf-feedstock/pull/1

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@ocefpaf We need to get netcdf-cxx4 ready first.

**Comment by kmuehlbauer** [comment]
@ocefpaf @jakirkham The problem with `hdf5` was that `conda` wasn't able to pull in the correct `hdf5` version at runtime.

So the dependency chain is as follows:

`libnetcdf` --> `netcdf-cxx4` --> `libradolan`

So the only place where hdf5 comes into play is `libnetcdf`. If all resolves properly, we do not need to hdf5-pin any more.

**Comment by ocefpaf** [comment]
Cool! I will update this PR once we get the others merged to see if we can drop the dependency over-specification.

**Comment by kmuehlbauer** [comment]
@ocefpaf This is blocked atm by `netcdf-cxx4`, which depends on fixing `ncxx4-config` at unidata: https://github.com/Unidata/netcdf-cxx4/pull/37

**Comment by kmuehlbauer** [comment]
@ocefpaf Can you restart the CI's here? Now, that the netcf-cxx4 thing is sorted out, we could try this here again.

**Comment by kmuehlbauer** [comment]
@ocefpaf nvm I just noticed that I could do it myself

**Comment by ocefpaf** [comment]
@kmuehlbauer this will need more thought. Does `libradolan` expects `ncxx4-config` to exist?

**Comment by kmuehlbauer** [comment]
@ocefpaf I'll have a look into this. Thanks!

**Comment by kmuehlbauer** [comment]
@ocefpaf Found it. We're creating wrong library names in https://github.com/conda-forge/netcdf-cxx4-feedstock. It should be `libnetcdf_c++4`, but we're creating `libnetcdf_cxx4`. Any clues?

**Comment by ocefpaf** [comment]
I guess the name changed in the update. I need to check netcdf-cxx4 code to be sure.

**Comment by kmuehlbauer** [comment]
Unidata is also producing `libnetcdf_c++4` ([travis-log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/130138115/log.txt)). Merkwürdig.

**Comment by ocefpaf** [comment]
I'll take a look. The only changes are the use of cmake and the update.
Maybe we should create a symlink?
Em 21/06/2016 06:50, "Kai Mühlbauer" notifications@github.com escreveu:

Unidata is also producing libnetcdf_c++4 (travis-log
https://s3.amazonaws.com/archive.travis-ci.org/jobs/130138115/log.txt).
Merkwürdig.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
https://github.com/conda-forge/libradolan-feedstock/pull/3#issuecomment-227393045,
or mute the thread
https://github.com/notifications/unsubscribe/AA6BL4tHSQSd4UdVP5LcEKbp93ubVM8zks5qN7PSgaJpZM4IqFLt
.

**Comment by kmuehlbauer** [comment]
> Unidata is also producing libnetcdf_c++4 (travis-log).

This was only half the truth. Obviously the `cmake` builds also at Unidata creates `libnetcdf_cxx4` whereas the `configure` build creates `libnetcdf_c++4`. 

@ocefpaf As you already suggested, a symlink should do the trick. Or just renaming as it mimicks the `configure` behaviour. WDYT?

**Comment by ocefpaf** [comment]
> WDYT?

Lets's symlink and cover both naming. Will do soon. Just trying to find a bug in conda-smithy now...

**Comment by ocefpaf** [comment]
@kmuehlbauer https://github.com/conda-forge/netcdf-cxx4-feedstock/pull/11 should fix the issue here.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,18 +11,18 @@ source:
...
-  number: 1
+  number: 2
   skip: true   # [win]
 
 requirements:
   build:
-    - boost 1.60.*
-    - netcdf-cxx4
     - cmake
+    - boost 1.61.*
```
Comment (ocefpaf) [inline recipe/meta.yaml]: @kmuehlbauer I updated the boost pinning here. Is that OK?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,18 +11,17 @@ source:
...
+    - netcdf-cxx4 4.3.*
   run:
-    - boost 1.60.*
-    - hdf5 1.8.15*
```
Comment (ocefpaf) [inline recipe/meta.yaml:24]: If we pin `netcdf-cxx4` we don't need to add the `hdf5` pin here b/c the right `libnetcdf`/`hdf5` will be downloaded.

Comment (kmuehlbauer) [inline recipe/meta.yaml:24]: > If we pin netcdf-cxx4 we don't need to add the hdf5 pin here b/c the right libnetcdf/hdf5 will be downloaded.

That had _historic_ reasons b/c that time this fixed some wrongly pulled hdf5. Now it won't be needed any more.

**Comment by kmuehlbauer** [comment]
[Travis Problems](https://www.traviscistatus.com/incidents/nf3tg08pqg2d), we have to wait.

**Comment by kmuehlbauer** [comment]
@ocefpaf We had this before, I think, https://github.com/conda-forge/netcdf-cxx4-feedstock/pull/5 This has to be some problem with libc++ / stdlibc++. We should have cmake and make output more verbos to see what is going on here.

**Comment by ocefpaf** [comment]
> @ocefpaf We had this before, I think, https://github.com/conda-forge/netcdf-cxx4-feedstock/pull/5 This has to be some problem with libc++ / stdlibc++. We should have cmake and make output more verbos to see what is going on here.

There is an ongoing conversation on `gdal`/`hdf5` feedstock about this. I don't have a Mac to follow that, but I guess we should read those threads before trying anything here.

**Comment by kmuehlbauer** [comment]
> There is an ongoing conversation on gdal/hdf5 feedstock about this. I don't have a Mac to follow that, but I guess we should read those threads before trying anything here.

We should also not forget, that `netcdf-cxx4` will have the same problems. Hopefully this will be sorted out soon.

**Comment by ocefpaf** [comment]
@kmuehlbauer success! :tada:

**Comment by jakirkham** [comment]
Looks clean AFAICT.

**Comment by kmuehlbauer** [comment]
@ocefpaf @jakirkham great news. Will check this soon. Now let's merge this.

**Pull request merged**